src/utils/compiler.test.ts

```diff
@@ -0,0 +1,26 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { describe, expect, test } from 'vitest';
+import { compile } from './compiler';
+
+const rootPath = `${__dirname}/../../`;
+const outputFile = `${__dirname}/../../dist/index.js`;
+const pathToIndex = '__mocks__/index.tsx';
+
+declare global {
+    interface Window {
+        index: { block: unknown; settings: unknown };
+    }
+}
+
+global.window = {} as any;
+
+describe('Compiler utils', async () => {
+    test('should provide a valid build with block and settings', async () => {
```

Comment:
Here are few more tests cases:
- Ensure `react` & `react-dom` are loaded from `window`
- Ensure `process.env.NODE_ENV` is correctly replaced
- Ensure when `process.env.NODE_ENV = 'production'` that code with `process.env.NODE_ENV === 'development'` is correctly removed (dead code elimination)

Reply:
ah definitely, I just pushed the first one 😉 more to come